subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DetachedConfigurationsIntegrationTest.groovy

```diff
@@ -65,4 +66,99 @@ class DetachedConfigurationsIntegrationTest extends AbstractIntegrationSpec {
         expect:
         run "checkDependencies"
     }
+
+    def "detached configurations may have project dependencies"() {
+        given:
+        settingsFile << "include 'other'"
+        buildFile << """
+            plugins {
+                id 'java-library'
+            }
+            
+            def detached = project.configurations.detachedConfiguration()
+            detached.dependencies.add(project.dependencies.create(project(':other')))
+           
+            task checkDependencies {
+                doLast {
+                    assert detached.resolvedConfiguration.getFirstLevelModuleDependencies().moduleName.contains('other')
+                    assert detached.resolvedConfiguration.resolvedArtifacts.collect { it.file.name }.contains("other.jar")
+                }
+            }
+        """
+
+        file("other/build.gradle") << """
+            plugins {
+                id 'java-library'
+            }
+        """
+
+        expect:
+        run "checkDependencies"
+    }
+
+    // This behavior will be removed in Gradle 9.0
+    @Deprecated
+    def "detached configurations may have project dependency on self"() {
+        given:
+        settingsFile << """
+            rootProject.name = 'test'
+        """
+
+        buildFile << """
+            plugins {
+                id 'java-library'
+            }
+            
+            def detached = project.configurations.detachedConfiguration()
+            detached.attributes.attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage, Usage.JAVA_RUNTIME))
+            detached.dependencies.add(project.dependencies.create(project))
+           
+            task checkDependencies {
+                doLast {
+                    assert detached.resolvedConfiguration.getFirstLevelModuleDependencies().moduleName.contains('test')
+                }
+            }
+        """
+
+        expect:
+        run "checkDependencies"
+    }
+
+    // This behavior will be removed in Gradle 9.0
+    @Deprecated
+    def "detached configurations can contain artifacts and resolve them during a self-dependency scenario"() {
+        given:
+        settingsFile << """
+            rootProject.name = 'test'
+        """
+
+        buildFile << """
+            plugins {
+                id 'java-library'
```

Comment:
Not sure we need to apply the java-library plugin for this one

Reply:
It's necessary, otherwise:

```Execution failed for task ':checkDependencies'.
> Could not resolve all dependencies for configuration ':detachedConfiguration1'.
   > Cannot query the value of task ':makeArtifact' property 'archiveFile' because it has no value available.```

Reply:
What if we apply the Java base plugin instead?